templates/layout/_header_bread.html.twig

```diff
@@ -12,14 +12,14 @@
     </div>
 {% elseif is_route_name_end_with('_subscribed') %}
     <div class="head-title">
-        <span>/</span><a href="{{ path('front_subscribed') }}">sub</a>
+        <span>/</span><a href="{{ '/sub/' }}">sub</a>
     </div>
 {% elseif is_route_name_end_with('_favourite') %}
     <div class="head-title">
-        <span>/</span><a href="{{ path('front_favourite') }}">fav</a>
+        <span>/</span><a href="{{ '/fav/' }}">fav</a>
     </div>
 {% elseif is_route_name_end_with('_moderated') %}
     <div class="head-title">
-        <span>/</span><a href="{{ path('front_moderated') }}">mod</a>
+        <span>/</span><a href="{{ '/mod/' }}">mod</a>
     </div>
 {% endif %}
```

Comment:
this is actually not working, because you made the `all|sub|fav|mod` a query parameter. Why is it not just part of the path?

Reply:
So, under my current code, either /mod/newest *or* /newest?filter=mod will work. They both have to work and do the same thing, and the reason is moderately complicated. Here's my thinking on it; if you can see a better way to handle this. I'm all ears.

So the issue is that before my changes, we had different routes for sub/fav/mod/all. You can see them all in config/kbin_routes/front.yaml. So the implication of that is, a lot of places we could use options_url() wouldn't work -- there was basically no general way to implement the functionality of switching one of the filters to a new setting but leaving the rest unchanged. You can see this on existing instances, e.g. switch the time filter to 6h, then switch to "moderated", and the time filter will reset back to "all time."

In the existing code there was a lot of custom hackery, manually checking the route to look for strings, and things like that, but it didn't really work even for the functionality that was already there and it wasn't going to be sustainable to try to keep making it worse to keep things working with newly-visible filters. So what I did was put it all through a single codepath: We store the filter values in $criteria, and we say that any given page that might get filtered always gets put through the same route, so that we try to always change filter values with options_url(). I didn't eliminate all of the special-cased stuff but I switched a substantial amount of different filtering stuff that went through different codepaths to all go through that codepath.

But, the cost of that unification was, because the URLs are structured with the sub/mod/fav at the beginning and optional, you *can't* have a single route entry that you can use for both e.g. /sub/newest and /newest. The cleanest way I could see to do it was to have it all go through either "root" or "front", and make both of them able to resolve the full set of filters including sub/mod/fav/all. It's just that in "root" it's a query parameter and in "front" it's part of the URL.

So yeah, it's a little weird, but the weird URL structure that causes that to happen is something we came in with, and I didn't see a better way to do it, and the end result is still an improvement on what came before IMO because we're not poking around doing string searches on routes etc, and we have a lot less duplicated code (e.g. the cut and pasted duplicate routes in EntryFrontController).

The other, cleaner, option would be just to break existing URLs and change the structure. As a general rule I'm reluctant to make a URL that used to work suddenly not work or be at a different location, but if people are cool with doing that then we can do this in a much cleaner way and drop the query param.

Reply:
So... I thought a little more after writing this comment. How do people feel about just changing the URL structure so that fav/mod/sub/all is later on in the URL?

I actually feel like breaking the /mod/ and friends URLs maybe isn't too bad, and the gain we'd have by moving to only one URL scheme would maybe be worth it.

The other option would be to move towards using query params for all this stuff; it's starting to get a little bit awkward to have so many parameters just in an exactly-ordered list of parts of the URL, where a key-value approach would be more robust, apply a lot more easily as we're trying to do this for other routes, and also easier to make sense of if someone's looking at the URL.

What do you think?

Reply:
I'd just make the fav/mod/sub/all stuff part of the URL path. It does not have to be in the first place, but it should be part of it and not a query parameter. If we want the breadcrumbs for it removed, I fine with it as well, though I'd like to keep it for fav/mod/sub. This is a very easy way to see that one is on the correct page, but I am open for discussion on this one. 

We could still keep redirects on `/sub` etc. to the new path if we'd want to

Reply:
Yeah, having it on the URL path in one consistent new way and having a redirect from the old (current) way sounds way better, you're right. I'll fix it to be that way + push a fix.

For breadcrumbs, let me share my thoughts in Matrix -- I think I may be causing some short-term breakage of breadcrumbs, but it's in service of going to a destination that I feel like would be way better in the long run. But if I share my overall thinking on it maybe we can come to something that's good for both the short and long term. And of course as always I may be missing or misunderstanding something.

Reply:
Just pushed an update that puts everything through the /$filter/$sortBy/etc route, and uses a redirect if anyone tries to access a /$sortby/etc path.

Should I put breadcrumbs for /$filter back into the nav bar? To me that's weird if the nav bar is adjusting as you're picking new filter params way down the page, but I'm fine with whatever way.

Reply:
Just put the breadcrumbs back. The breadcrumb and top bar links probably want some testing but I poked at them some and it seem like they're working to me now.